xcp_d/interfaces/prepostcleaning.py

```diff
@@ -276,10 +278,15 @@ def _run_interface(self, runtime):
         # Read in custom confounds file (if any) and bold file to be censored
         bold_file_uncensored = nb.load(self.inputs.in_file).get_fdata()
         if self.inputs.custom_confounds:
-            custom_confounds_tsv_uncensored = pd.read_csv(
-                self.inputs.custom_confounds,
-                header=None,
-            )
+            if os.path.exists(self.inputs.custom_confounds):
```

Comment:
Why check if the file exists? What is the value of `self.inputs.custom_confounds` if it's not a file?

Reply:
For some reason, just if self.inputs.custom_confounds was present then it would fail - is there another way to fix that?

Reply:
I think you might need to add an else clause and print out the type and value when it isn't a file. If it's a "None" string, then there's probably a way to check that earlier on.

Reply:
[`nipype.interfaces.base.traits_extension.isdefined()`](https://nipype.readthedocs.io/en/latest/api/generated/nipype.interfaces.base.traits_extension.html#nipype.interfaces.base.traits_extension.isdefined) might be the thing to use. It should check if an input is undefined or not.

Reply:
will try!